src/newcharacter.cpp

```diff
@@ -1892,6 +1892,20 @@ tab_direction set_profession( avatar &u, points_left &points,
             }
             const int netPointCost = sorted_profs[cur_id]->point_cost() - u.prof->point_cost();
             u.prof = &sorted_profs[cur_id].obj();
+            // Remove pre-selected traits that conflict
+            // with the new profession's traits
+            for( const trait_id &new_trait : u.prof->get_locked_traits() ) {
+                if( u.has_conflicting_trait( new_trait ) ) {
+                    for( const trait_id &suspect_trait : u.get_mutations() ) {
+                        if( are_conflicting_traits( new_trait, suspect_trait ) ) {
+                            u.toggle_trait( suspect_trait );
+                            points.trait_points += suspect_trait->points;
+                            popup( _( "Your trait %s has been removed since it conflicts with the %s's %s trait." ),
```

Comment:
```suggestion
                            popup( _( "Your trait %1$s has been removed since it conflicts with the %2$s's %3$s trait." ),
```
These are so the translators can rearrange words as needed - Spanish, for instance, would do the last two in reverse order (%3$s de %2$s). (@BrettDong, can you confirm if I've done it right and if there's anything else needed? Thanks!)

Reply:
You're right!